src/mesh/Router.cpp

```diff
@@ -257,10 +257,9 @@ ErrorCode Router::send(meshtastic_MeshPacket *p)
             return encodeResult; // FIXME - this isn't a valid ErrorCode
         }
 
-        if (moduleConfig.mqtt.enabled) {
-            LOG_INFO("Should encrypt MQTT?: %d\n", moduleConfig.mqtt.encryption_enabled);
-            if (mqtt)
-                mqtt->onSend(*p, *p_decoded, chIndex);
+        // Only publish to MQTT if we're the original transmitter of the packet
+        if (moduleConfig.mqtt.enabled && p->from == nodeDB.getNodeNum() && mqtt) {
```

Comment:
Just confirming, we would still "rebroadcast" a message to MQTT received from someone else transmitting on lora?

Reply:
Yes, in this case it's handled upon receiving the packet via LoRa: 
https://github.com/meshtastic/firmware/blob/834490175d312dd7c4a2d6049185f030fffe85a3/src/mesh/Router.cpp#L454-L456